cirq-google/cirq_google/workflow/quantum_runtime.py

```diff
@@ -180,26 +212,38 @@ def execute(
         # coverage: ignore
         raise ValueError("Please provide a non-empty `base_data_dir`.")
 
-    shared_rt_info = SharedRuntimeInfo(run_id=run_id)
+    sampler = rt_config.processor.get_sampler()
+    device = rt_config.processor.get_device()
+
+    shared_rt_info = SharedRuntimeInfo(
+        run_id=run_id,
+        device=device,
+    )
     executable_results = []
 
     saver = _FilesystemSaver(base_data_dir=base_data_dir, run_id=run_id)
     saver.initialize(rt_config, shared_rt_info)
 
-    sampler = rt_config.processor.get_sampler()
     logger = _PrintLogger(n_total=len(executable_group))
     logger.initialize()
+
+    rs = np.random.RandomState(rt_config.random_seed)
+    exe: QuantumExecutable
```

Comment:
Does this declaration mess with the below loop ?

Reply:
this is how you do type annotations for loop variables

Reply:
Do we need this annotation? Wouldn't the type of `exe` get inferred automatically based on the type of `executable_group` which we iterating on?

Reply:
pycharm wasn't picking it up